com.unity.render-pipelines.high-definition/Runtime/Lighting/AtmosphericScattering/Fog.cs

```diff
@@ -51,28 +47,50 @@ public class Fog : VolumeComponent
+        public FogControlParameter fogControlMode = new FogControlParameter(FogControl.Balance);
         /// <summary>Resolution of the volumetric buffer (3D texture) along the X and Y axes relative to the resolution of the frame buffer.</summary>
         [Tooltip("Resolution of the volumetric buffer, along the x-axis and y-axis, relative to the resolution of the frame buffer.")]
-        public ClampedFloatParameter screenResolutionPercentage = new ClampedFloatParameter((1.0f/8.0f) * 100, (1.0f/16.0f) * 100, 100);
+        public ClampedFloatParameter screenResolutionPercentage = new ClampedFloatParameter(optimalFogScreenResolutionPercentage, minFogScreenResolutionPercentage, maxFogScreenResolutionPercentage);
         /// <summary>Number of slices of the volumetric buffer (3D texture) along the camera's focal axis.</summary>
         [Tooltip("Number of slices of the volumetric buffer (3D texture) along the camera's focal axis.")]
-        public ClampedIntParameter volumeSliceCount = new ClampedIntParameter(64, 1, 1024);
+        public ClampedIntParameter volumeSliceCount = new ClampedIntParameter(64, 1, maxFogSliceCount);
+        /// <summary>The slider controls in a linear way the cost and quality of the volumetric fog. The value 0 being the least expensive and 1 the highest quality.</summary>
+        [Tooltip("The slider controls in a linear way the cost and quality of the volumetric fog. The value 0 being the least expensive and 1 the highest quality.")]
+        public ClampedFloatParameter volumetricFogBudget = new ClampedFloatParameter(0.25f, 0.0f, 1.0f);
+        /// <summary>This parameters controls how the budget is shared between Screen (XY) and Depth (Z) resolutions. The value 0 means that all of the budget is used for the XY resolution, which will reduce aliasing, but will increase noise. On the other hand a value of 1 will put more emphasis on the Z resolution reducing the noise, but increasing the aliasing. The parameters allows linear interpolation between the two configurations.</summary>
+        [Tooltip("This parameters controls how the budget is shared between Screen (XY) and Depth (Z) resolutions. The value 0 means that all of the budget is used for the XY resolution, which will reduce aliasing, but will increase the noise. On the other hand a value of 1 will put more emphasis on the Z resolution reducing the noise, but increasing the aliasing. The parameters allows linear interpolation between the two configurations.")]
+        public ClampedFloatParameter resolutionDepthRatio = new ClampedFloatParameter(0.5f, 0.0f, 1.0f);
 
-        /// <summary>Applies a blur to smoothen the volumetric lighting output.</summary>
-        [Tooltip("Applies a blur to smoothen the volumetric lighting output.")]
-        public BoolParameter filter = new BoolParameter(false);
+        /// <summary>Includes or excludes non-directional light types into evaluating the volumetric fog.</summary>
```

Comment:
Indicates whether Unity includes or excludes non-directional light types when it evaluates the volumetric fog. Including non-directional lights increases the resource intensity of the effect.